### PR TITLE
[WIP] Add detailed guide on Clock

### PR DESCRIPTION
**Pull request by JoelQ**
This adds a more detailed guide on the clock exercise with discussion
points and code samples.

With hamming, we'd discussed having a separate high-level bullet-list
style overview doc and a standalone deeper-dive guide. Not sure if we
want to do this here or try to merge this guide into the clock `README`
file.

### CODE REVIEW

**Comment by iHiD** [comment]
I'll fold this into the README.

File: tracks/ruby/exercises/clock/GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+### Modeling objects with a single quantity versus separate minutes and hours
+
+This makes the math easier. Storing hours and minutes separately is effectively
+caching a calculation. It's usually done because the student is modeling human
+groupings directly. Help student realize that "time elapsed since midnight" is a
+single quantity, even though humans represent it as two quantities for ease of
+reading.
+
+### Rerturn new instances from `+` and `-`
```
Comment (elle) [inline tracks/ruby/exercises/clock/GUIDE.md]: *Return

File: tracks/ruby/exercises/clock/GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  MINUTES_PER_HOUR = 60
+  MINUTES_PER_DAY = 1440
+
+  def initialize(hour: 0, minute: 0)
+    @total_minutes = ((hour * MINUTES_PER_HOUR) + minute) % MINUTES_PER_DAY
+  end
+
+  def to_s
+    hours = total_minutes / MINUTES_PER_HOUR
+    minutes = total_minutes % MINUTES_PER_HOUR
```
Comment (elle) [inline tracks/ruby/exercises/clock/GUIDE.md]: Would you pull this two variables out to their own methods?

Comment (JoelQ) [reply tracks/ruby/exercises/clock/GUIDE.md]: I like it 👍

**Comment by JoelQ** [comment]
Hi @elle 👋

**Pull request merged**